libs/core/src/main/java/org/elasticsearch/bootstrap/JarHell.java

```diff
@@ -255,6 +255,10 @@ public static void checkJavaVersion(String resource, String targetVersion) {
     }
 
     private static void checkClass(Map<String, Path> clazzes, String clazz, Path jarpath) {
+        if (clazz.equals("module-info") || clazz.endsWith(".module-info")) {
```

Comment:
Why are there 2 variants? Shouldn't the class name always be module-info?

Reply:
That's correct, but it can live in a package and users might put it in the same package e.x. `org.elasticsearch.module-info`